src/privatesend-util.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2014-2017 The Dash Core developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#include "privatesend-util.h"
+
+KeyHolder::KeyHolder(CWallet* pwallet)
+        :reserveKey(pwallet)
+{
+        reserveKey.GetReservedKey(pubKey, false);
+}
+
+void KeyHolder::KeepKey()
+{
+    reserveKey.KeepKey();
+}
+
+void KeyHolder::ReturnKey()
+{
+    reserveKey.ReturnKey();
+}
+
+CScript KeyHolder::GetScriptForDestination()
+{
+    return ::GetScriptForDestination(pubKey.GetID());
+}
+
+
+KeyHolderPtr KeyHolderStorage::AddKey(CWallet* pwallet)
```

Comment:
Yes, bad suggestion; returning CScript is a hack.

The better solution is to return a (constant) reference.